test/scientist/experiment_test.rb

```diff
@@ -409,6 +409,26 @@ def @ex.raised(op, exception)
       assert_raises(Scientist::Experiment::MismatchError) { @ex.run }
     end
 
+    describe "#raise_on_mismatches?" do
+      it "raises when there is a mismatch if the experiment instance's raise on mismatches is enabled" do
+        Fake.raise_on_mismatches = false
+        @ex.raise_on_mismatches = true
+        @ex.use { "fine" }
+        @ex.try { "not fine" }
+
+        assert_raises(Scientist::Experiment::MismatchError) { @ex.run }
+      end
+
+      it "doesn't raise when there is a mismatch if the experiment instance's raise on mismatches is disabled" do
+        Fake.raise_on_mismatches = true
+        @ex.raise_on_mismatches = false
+        @ex.use { "fine" }
+        @ex.try { "not fine" }
+
+        assert_equal "fine", @ex.run
+      end
+    end
+
```

Comment:
Would these test cases for class level raise_on_mismatches suffice? https://github.com/tyamagu2/scientist/blob/raise_if_mismatches_per_instance/test/scientist/experiment_test.rb#L382:L396